test/IntegrationTests/test_listfile.py

```diff
@@ -0,0 +1,147 @@
+import unittest
+import requests
+import logging
+
+logging.basicConfig(
+    level=logging.INFO,
+    format='%(asctime)s - %(levelname)s - %(message)s'
+)
+logger=logging.getLogger(__name__)
+
+class TestListFileAPI(unittest.TestCase):
+    BASE_URL="http://localhost:8000"
+
+    def setUp(self):
+        logger.info("=== Starting setup process ===")
+
+        add_customer_url=f"{self.BASE_URL}/addcustomer"
+        logger.info(f"Requesting new customer from:{add_customer_url}")
+
+        response=requests.post(add_customer_url)
+        self.assertEqual(response.status_code, 200, "Failed to create customer")
+
+        customer_data=response.json()
+        self.customer_guid=customer_data.get("customer_guid")
+        logger.info(f"Received valid customer_guid: {self.customer_guid}")
+
+        # upload a test file here to ensure that there are files to list
+        upload_file_url=f"{self.BASE_URL}/uploadFile"
+        test_file_data ={"customer_guid": self.customer_guid}
+        test_file=("file",("testfile.txt",b"Hello,Python","text/plain"))
+        file_response = requests.post(upload_file_url, data=test_file_data, files=[test_file])
+
+        self.assertEqual(file_response.status_code, 200, "File upload failed during setup")
+
+        logger.info("=== Setup process completed ===")
+
+    def test_list_files_success(self):
+        logger.info("Testing file listing for a valid customer_guid")
+
+        list_files_url=f"{self.BASE_URL}/listfiles"
+        params={"customer_guid": self.customer_guid}
+        response=requests.get(list_files_url, params=params)
+
+        # Verify the HTTP response status code
+        if response.status_code==200:
+            logger.info("Response status code is valid and matches the expected value (200). File listing successful.")
+        else:
+            logger.error(f"Unexpected status code: {response.status_code}. Expected: 200.")
+            raise AssertionError(f"Expected status code 200 but got {response.status_code}")
+
+        # Assert the response
+        self.assertEqual(response.status_code, 200, "Failed to list files for valid customer_guid")
+        self.assertIn("files", response.json(), "No 'files' key returned in response")
+
```

Comment:
Verify the names of the file here

Try creating a separate customer_guid and upload the files local to the function instead of setup function